### PR TITLE
json handler: add adaptive batching support

### PR DESCRIPTION
**Pull request by hrmthw**
TODOs:

* add warning about the breaking change of json handler
* add `Bento-Multi-Input-Flag`
* update examples

### CODE REVIEW

File: bentoml/handlers/json_handler.py [hunk 1]
```diff
@@ -45,8 +50,42 @@ def handle_request(self, request, func):
...
+        responses = [bad_resp] * len(requests)
+
+        for i, request in enumerate(requests):
+            try:
+                raw_str = request.data
+                parsed_json = json.loads(raw_str)
+                instances_list[i] = parsed_json
+            except (json.JSONDecodeError, UnicodeDecodeError):
+                responses[i] = SimpleResponse(400, None, "not a valid json input")
+            except Exception:  # pylint: disable=broad-except
```
Comment (parano) [inline bentoml/handlers/json_handler.py]: any specific reason that we need this? I think the API sever side does a broad except and will handle unexpected errors

File: bentoml/handlers/json_handler.py [hunk 2]
```diff
@@ -45,8 +50,42 @@ def handle_request(self, request, func):
...
+            raise ValueError(
+                "The return value with JsonHandler must be list of jsonable objects, "
+                "and have same length as the inputs."
+            )
+
+        results = [merged_result[s] for s in slices]
+
+        for i, result in enumerate(results):
+            result_str = api_func_result_to_json(result)
+            responses[i] = SimpleResponse(200, dict(), result_str)
```
Comment (parano) [inline bentoml/handlers/json_handler.py]: isn't this going to rewrite the  `SimpleResponse(400, None, "not a valid json input")` above?

Comment (parano) [reply bentoml/handlers/json_handler.py]: maybe in line 58 initialize it as:
```
        responses = [None] * len(requests)
```
and here you can do something like:
```
if responses[I] is None:
   responses[i] = SimpleResponse(200, dict(), result_str)
```

Comment (hrmthw) [reply bentoml/handlers/json_handler.py]: Oh, I see. I'll make sure these handlers are as robust as DataframeHandler did in later commits.

File: bentoml/handlers/json_handler.py (hunk 1 above)
Comment (hrmthw) [reply bentoml/handlers/json_handler.py]: To catch the error so that other normal requests could be properly returned.

**Comment by hrmthw** [comment]
@parano updated

File: bentoml/handlers/json_handler.py (hunk 1 above)
Comment (parano) [reply bentoml/handlers/json_handler.py]: 👍 got it

**Pull request merged**